lexicon/providers/corenetworks.py

```diff
@@ -0,0 +1,280 @@
+"""Module provider for DNS Simple"""
+from __future__ import absolute_import
+import json
+import hashlib
+import logging
+import time
+import os
+
+import requests
+from lexicon.providers.base import Provider as BaseProvider
+
+LOGGER = logging.getLogger(__name__)
+
+NAMESERVER_DOMAINS = ['core-networks.de', 'core-networks.eu', 'core-networks.com']
+
+def provider_parser(subparser):
+    """Configure provider parser for Core Networks"""
+    subparser.add_argument(
+        "--auth-username", help="specify login for authentication")
+    subparser.add_argument(
+        "--auth-password", help="specify password for authentication")
+    subparser.add_argument(
+        "--auth-file", help="specify location for authentication file")
+
+class Provider(BaseProvider):
+    """Provider class for Core Networks"""
+    def __init__(self, config):
+        super(Provider, self).__init__(config)
+        self.domain_id = None
+        self.account_id = None
+        self.token = None
+        self.expiry = None
+        self.modified = False
+        self.auth_file = { 'token': None, 'expiry': None }
+        # Core Networks enforces a limit on the amount of logins per minute.
+        # As the token is valid for 1 hour it's sensible to store it for
+        # later usage.
+        self.auth_file_path = self._get_provider_option('auth_file') or '/tmp/corenetworks_auth.json'
```

Comment:
I think `/var/lib/lexicon/corenetworks_auth.json` would be a more appropriate location.

Reply:
I thought about such directories, too, but during development I worked in a virtual environment without write permissions to `/var/lib`. Ideally, there should be some global variable like `config['data_dir']`. But I'm new to Python and new to lexicon.

Reply:
Indeed /var/lib may not be writable, if you are not root. There is not such thing like a `data_dir` for now, since most providers just use what they get from environment variable, a given Lexicon config YAML file or CLI flags.

But here it is for caching the credentials, which is quite different. Since it contains sensitive data, I would put it in the HOME of the user, with a fallback in TMP directory if HOME does not exist. HOME can get in platform agnostic way with `os.path.expanduser("~")`, TMP similarly with `tempfile.gettempdir()` (which on Windows will be in fact a TMP directory inside the HOME user).

But the really important thing, no matter the location, is that the file has permission `0700`. Nobody expect current user should be able to see that file.

Reply:
In method _commit_auth_file I assign a permission mode of 0600 as it does not need to be executable.